### PR TITLE
Adding unit tests- powervs

### PR DESCRIPTION
**Pull request by Amulyam24**
**What this PR does / why we need it**:
This PR adds unit tests for PowerVS

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Specify your PR reviewers**:

/cc @gyliu513 @jichenjc @xunpan @mkumatag 

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by mkumatag** [comment]
cc @Karthik-K-N

**Comment by Amulyam24** [comment]
I signed it

File: controllers/ibmpowervscluster_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	"github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/api/v1alpha4"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2/klogr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("IBMPowerVSClusterReconciler", func() {
+	BeforeEach(func() {})
+	AfterEach(func() {})
```
Comment (mkumatag) [inline controllers/ibmpowervscluster_controller_test.go:33]: do we need these if empty?

File: controllers/ibmpowervsmachine_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	"github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/api/v1alpha4"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2/klogr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("IBMPowerVSMachineReconciler", func() {
+	BeforeEach(func() {})
+	AfterEach(func() {})
```
Comment (mkumatag) [inline controllers/ibmpowervsmachine_controller_test.go:34]: same as above

**Review by mkumatag — CHANGES_REQUESTED** [review]
lgtm for the initial commit, need to explore the options for mocking the client and cover more scenarios,

some minor nits

File: controllers/ibmpowervscluster_controller_test.go (hunk 1 above)
Comment (Amulyam24) [reply controllers/ibmpowervscluster_controller_test.go:33]: No, we can skip them.

File: controllers/ibmpowervscluster_controller_test.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import (
+	"context"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/api/v1alpha4"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2/klogr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
```
Comment (mkumatag) [inline controllers/ibmpowervscluster_controller_test.go:28]: minor import formatting

```suggestion
	"context"

	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
	"k8s.io/klog/v2/klogr"
	ctrl "sigs.k8s.io/controller-runtime"
	"sigs.k8s.io/controller-runtime/pkg/client"
	
	"github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/api/v1alpha4"
	
	. "github.com/onsi/ginkgo"
	. "github.com/onsi/gomega"
```

File: controllers/ibmpowervsmachine_controller_test.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+	"context"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/api/v1alpha4"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2/klogr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
```
Comment (mkumatag) [inline controllers/ibmpowervsmachine_controller_test.go:29]: same suggestion as above

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: controllers/ibmpowervscluster_controller_test.go (hunk 3 above)
Comment (Amulyam24) [reply controllers/ibmpowervscluster_controller_test.go]: Updated @mkumatag

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**